runtime/queries/gleam/highlights.scm

```diff
@@ -56,14 +64,37 @@
 
 (attribute_value (identifier) @constant)
 
+; =========
+; Types
+; =========
+
 ; Type names
 (remote_type_identifier) @type
 (type_identifier) @type
 
+; Generic types
+[
+  ; in `pub type Dict(key, value)` this is `key` and `value`
+  (type_parameter)
+  ; in `pub fn size(dict: Dict(key, value)) -> Int` this is `key` and `value`
+  (type_var)
+] @type
+
 ; Data constructors
 (constructor_name) @constructor
 
+; built-ins
+((constructor_name) @constant.builtin
+  (#any-of? @constant.builtin "False" "True"))
+((constructor_name) @constant.builtin
+  (#any-of? @constant.builtin "Nil"))
+((constructor_name) @type.enum.variant.builtin
+  (#any-of? @type.enum.variant.builtin "Ok" "Error" "Some" "None"))
+
+; =========
 ; Literals
+; =========
+
 (string) @string
 ((escape_sequence) @warning
```

Comment:
Not exactly related to this PR but I noticed that this escape_sequence pattern should be swapped with the one below it to fix the precedence